### PR TITLE
Add env variables

### PR DESCRIPTION
**Pull request by Stern-Ritter**

### CODE REVIEW

File: cmd/agent/main.go [hunk 1]
```diff
@@ -1,3 +1,70 @@
...
+	httpClient := resty.New()
+	cache := storage.NewAgentMemCache(model.SupportedGaugeMetrics, model.SupportedCounterMetrics)
+	monitor := model.Monitor{}
+	random := utils.NewRandom()
+
+	run(httpClient, &cache, &monitor, &random, config)
+}
+
+func run(httpClient *resty.Client, cache storage.AgentCache, monitor *model.Monitor, random *utils.Random, config config.AgentConfig) {
+	var wg sync.WaitGroup
```
Comment (arttet) [inline cmd/agent/main.go:39]:
```suggestion
	wg := &sync.WaitGroup{}
```

File: cmd/agent/main.go [hunk 2]
```diff
@@ -1,3 +1,70 @@
...
+		tasks.SendMetrics(httpClient, config.SendMetricsURL, config.SendMetricsEndPoint, cache)
+	}
+
+	tasks.SetInterval(ctx, &wg, updateMetricsTask, time.Duration(config.UpdateMetricsInterval)*time.Second)
+	tasks.SetInterval(ctx, &wg, sendMetricsTask, time.Duration(config.SendMetricsInterval)*time.Second)
+
+	wg.Wait()
+}
+
+func getConfig(c config.AgentConfig) (config.AgentConfig, error) {
```
Comment (arttet) [inline cmd/agent/main.go:58]: SUGGESTION: it would be better if it is in `flags.go`

File: cmd/server/main.go [hunk 3]
```diff
@@ -1,3 +1,46 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	storage := storage.NewServerMemStorage()
+	run(&storage, config)
+}
+
+func run(storage *storage.ServerMemStorage, config config.ServerConfig) {
+	router := chi.NewRouter()
```
Comment (arttet) [inline cmd/server/main.go:25]: SUGGESTION: it would be better if it's another function

File: cmd/server/main.go [hunk 4]
```diff
@@ -1,3 +1,46 @@
...
+	router.Get("/", handlers.GetMetricsHandler(storage))
```
Comment (arttet) [inline cmd/server/main.go:26]: CHANGE REQUEST: please split your application by layers.

File: internal/config/config.go [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+type AgentConfig struct {
+	SendMetricsURL        string `env:"ADDRESS"`
+	SendMetricsEndPoint   string
+	UpdateMetricsInterval int `env:"POLL_INTERVAL"`
+	SendMetricsInterval   int `env:"REPORT_INTERVAL"`
+}
+
+func (c *AgentConfig) ParseFlags() error {
+	flag.StringVar(&c.SendMetricsURL, "a", "localhost:8080", "address and port to run server in format <host>:<port>")
```
Comment (arttet) [inline internal/config/config.go:17]: CHANGE REQUEST: this is an application specific code.

File: internal/transport/handlers.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+	"sort"
+	"strings"
+
+	"github.com/Stern-Ritter/metrics-and-alerting-service/internal/errors"
+	"github.com/Stern-Ritter/metrics-and-alerting-service/internal/model"
+	"github.com/Stern-Ritter/metrics-and-alerting-service/internal/storage"
+	"github.com/go-chi/chi"
+)
+
+func UpdateMetricHandler(storage storage.ServerStorage) http.HandlerFunc {
```
Comment (arttet) [inline internal/transport/handlers.go:15]: CHANGE REQUEST: please use handler methods

File: internal/transport/handlers.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+		gauges, counters := storage.GetMetrics()
+		body := getMetricsList(gauges, counters)
+
+		res.Header().Set("Content-type", "text/html")
+		res.WriteHeader(http.StatusOK)
+		io.WriteString(res, body)
+	}
+}
+
+func getMetricsList(gauges map[string]model.GaugeMetric, counters map[string]model.CounterMetric) string {
```
Comment (arttet) [inline internal/transport/handlers.go:59]:
```suggestion
func getMetricsString(gauges map[string]model.GaugeMetric, counters map[string]model.CounterMetric) string {
```

File: internal/transport/handlers_test.go [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+	"github.com/Stern-Ritter/metrics-and-alerting-service/internal/errors"
+	"github.com/Stern-Ritter/metrics-and-alerting-service/internal/model"
+	"github.com/Stern-Ritter/metrics-and-alerting-service/internal/storage"
+	"github.com/go-resty/resty/v2"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+type MockStorage struct {
```
Comment (arttet) [inline internal/transport/handlers_test.go:18]: SUGGESTION: https://github.com/uber-go/mock

**Review by arttet — CHANGES_REQUESTED** [review]
Please add 
- https://about.codecov.io/blog/getting-started-with-code-coverage-for-golang/
- https://github.com/golangci/golangci-lint

File: cmd/agent/main.go (hunk 1 above)
Comment (Stern-Ritter) [reply cmd/agent/main.go]: done

File: cmd/server/main.go (hunk 3 above)
Comment (Stern-Ritter) [reply cmd/server/main.go]: done

File: cmd/server/main.go (hunk 4 above)
Comment (Stern-Ritter) [reply cmd/server/main.go]: done

File: internal/config/config.go (hunk 5 above)
Comment (Stern-Ritter) [reply internal/config/config.go]: done

File: internal/transport/handlers.go (hunk 6 above)
Comment (Stern-Ritter) [reply internal/transport/handlers.go]: done

File: internal/transport/handlers.go (hunk 7 above)
Comment (Stern-Ritter) [reply internal/transport/handlers.go]: done

File: internal/transport/handlers_test.go (hunk 8 above)
Comment (Stern-Ritter) [reply internal/transport/handlers_test.go]: done

**Review by arttet — APPROVED** [review]

**Pull request merged**